### PR TITLE
[SYCL-Graph][Doc] Graph nodes cannot depend on events from outside th…

### PR DESCRIPTION
**Pull request by guoyejun**
…e graph

### CODE REVIEW

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc [hunk 1]
```diff
@@ -1095,6 +1095,11 @@ will throw synchronously with error code `invalid`.
 - Waiting on a queue in the recording state is an error and will throw
 synchronously with error code `invalid`.
 
+Events returned from queue submissions when a queue is not in the recording
```
Comment (EwanC) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc:1098]: This is very similar to the first paragraph, can we just change the first paragraph encompass both situation. Something like:

"For queue submissions that are being recorded to a modifiable `command_graph`, the only events that can be used as parameters to `handler::depends_on()`, or as dependent events for queue shortcuts like `queue::parallel_for()`, are events that have been returned from queue submissions recorded to the same modifiable `command_graph`"

Comment (guoyejun) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc:1098]: for this change, we'd also add something (need to refine the sentence) in one of the sub-items which are supposed to be in queue in normal state if we don't change the first paragraph.

- Waiting on an event returned from a queue submission recorded to a graph will throw synchronously with error code `invalid`, for the case waiting in another queue or waiting in the same queue but not in the same record state ...

Comment (guoyejun) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc:1098]: > "For queue submissions that are being recorded to a modifiable `command_graph`, the only events that can be used as parameters to `handler::depends_on()`, or as dependent events for queue shortcuts like `queue::parallel_for()`, are events that have been returned from queue submissions recorded to the same modifiable `command_graph`"

just think about it again, these words only describes the case that what events can be used for the queue in recording state, this is also what my PR wants to add.  But the case in the original first paragraph would be missed, the events generated from the queue in recording state, could not be used outside of the recording state.

so, my idea is to use your paragraph to replace my change, while keeping the original first paragraph unchanged.

Comment (EwanC) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]: Having two paragraphs that describe almost identical valid usage I find confusing, a good goal would be to trying simplify the wording rather than add to it. I think this section should be structured as a paragraph describing what the valid events that can be used as dependencies in modifiable graph are, followed by restrictions on usage of events returned from graph recording.

> Waiting on an event returned from a queue submission recorded to a graph will throw synchronously with error code invalid, for the case waiting in another queue or waiting in the same queue but not in the same record state ...

"Waiting" here is intended to mean a blocking host-side wait(), rather than as a event dependency. So I think that shouldt should be clarified. I also realize the the point about queue waits isn't related to events, so maybe should have it's own section. I propose this text:

```
==== Event Limitations

For queue submissions that are being recorded to a modifiable `command_graph`,
the only events that can be used as parameters to `handler::depends_on()`, or
as dependent events for queue shortcuts like `queue::parallel_for()`, are events
that have been returned from queue submissions recorded to the same modifiable
`command_graph`.

Other limitations on the events returned from a submission to a queue in the
recording state are:

- Calling `event::get_info<info::event::command_execution_status>()` or
`event::get_profiling_info()` will throw synchronously with error code `invalid`.

- A blocking host-side wait on the event will throw synchronously with error
code `invalid`.

==== Queue Limitations

A blocking host-side wait on a queue in the recording state is an error and will
throw synchronously with error code `invalid`.

Comment (guoyejun) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]: nice wording!

> ... events that have been returned from queue submissions recorded to the same modifiable `command_graph`.

Since the command_graph could be modified again, and so I plan to change it to:
... recorded to the same modifiable `command_graph` or a new command_graph modified from the same modifiable `command_graph`.

> A blocking host-side wait

Is there a host-side wait unblocking?  If no, we may remove the 'blocking' here.

**Comment by guoyejun** [comment]
I changed a bit in the new commit to cover all the cases for event in the below figure, please review @EwanC , thanks.
image

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc [hunk 2]
```diff
@@ -1080,20 +1080,27 @@ they do in non-recording mode.
...
 
 - Calling `event::get_info<info::event::command_execution_status>()` or
-`event::get_profiling_info()` on an event returned from a queue submission
-recorded to a graph will throw synchronously with error code `invalid`.
+`event::get_profiling_info()` will throw synchronously with error code `invalid`.
+
+- A host-side wait on the event will throw synchronously with error
+code `invalid`.
+
+- the event cannot be used outside of the recording scope.
```
Comment (EwanC) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc:1098]: We should say how this will fail.
```suggestion
- Using the event outside of the recording scope will throw synchronously with error code
`invalid`.
```

**Review by EwanC — APPROVED** [review]
LGTM 🙂
